sympy/sets/tests/test_fancysets.py

```diff
@@ -295,6 +295,12 @@ def test_Range_set():
         assert r.inf == rev.inf and r.sup == rev.sup
         assert r.step == -rev.step
 
+    # test symbolic Range
+    n = symbols('n', integer=True)
+    r = Range(1, n)
```

Comment:
what happens if you call `.doit()`?

Reply:
It remains un-evaluated, if any of `start`, `stop`, `step` is a symbol.
**Code**
```python
from sympy import Range, symbols
n = symbols('n', integer=True)
r1, r2, r3 = Range(n, 10), Range(1, n), Range(1, 5, n)
print(r1.doit(), r2.doit(), r3.doit())
```
**Output**
```python
Range(n, 10, 1) Range(1, n, 1) Range(1, n*ceiling(4/n) + 1, n)
```

Reply:
As is apparent, things don't go well, when `step` is symbolic. It needs to be fixed before moving forward.
**Code**
```python
from sympy import Range, symbols
n = symbols('n', integer=True)
r3 = Range(1, 5, n)
for i in range(-2, 7):
    if i == 0:
        continue
    print(i, r3.subs(n, i))
``` 
**Output**
```python
-2 Range(0, 0, 1)
-1 Range(0, 0, 1)
1 Range(1, 5, 1)
2 Range(1, 5, 2)
3 Range(1, 7, 3)
4 Range(1, 5, 4)
5 Range(1, 6, 5)
6 Range(1, 7, 6)
```

Reply:
I think, my changes haven't done anything wrong. On running something similar on `master`, the following happens,
**Code**
```python
from sympy import Range, symbols
for i in range(-2, 7):
    if i == 0:
        continue
    print(i, Range(1, 5, i))
```
**Output**
```python
-2 Range(0, 0, 1)
-1 Range(0, 0, 1)
1 Range(1, 5, 1)
2 Range(1, 5, 2)
3 Range(1, 7, 3)
4 Range(1, 5, 4)
5 Range(1, 6, 5)
6 Range(1, 7, 6)
```
As it appears, the output is same. Well I will say, last few outputs are examples of `garbage-in-garbage-out`, may be it doesn't needs a fix. Or if we want to fix and want to it to behave like python `range` something is to be done regarding the following code,
```python
if not start.is_infinite:
            ref = start if start.is_finite else stop
            n = ceiling((stop - ref)/step)
            if n <= 0:
                # null Range
                start = end = S.Zero
                step = S.One
            else:
                end = ref + n*step
        return Basic.__new__(cls, start, end, step)
```
This can be fixed by,
```python
if not start.is_infinite:
            ref = start if start.is_finite else stop
            n = ceiling((stop - ref)/step)
            if n <= 0:
                # null Range
                start = end = S.Zero
                step = S.One
            else:
                end = ref + n*step
        if end > stop:
            end = stop 
        return Basic.__new__(cls, start, end, step)
```
PS:
That's how python `range` behaves,
```python
Python 3.6.7 (default, Oct 22 2018, 11:32:17) 
[GCC 8.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> list(range(1, 5, 7))
[1]
>>> range(1, 5, 7)
range(1, 5, 7)
```

Reply:
I have observed that there is a difference between Python 3's `range` and Python 2's xrange. Earlier, `Range` was behaving like, `xrange` irrespective of the python version. I have made the changes to add the version check before returning the check. See,  https://github.com/sympy/sympy/pull/17146/files#diff-c1c17bdd28aafd400bf9415c7d048382R566

Reply:
This isn't garbage. Range canonicalises its args so that any two Ranges that would give the same numbers will have the same args. This is useful for comparing them with `==`.

Reply:
I will check if my changes are affecting canonicalization of the input parameters and will do the necessary changes. It would be helpful if you can provide any example of such an affect due to my changes. Thanks.

Reply:
Since the previous tests are not passing, so canonical outputs are still being generated. I will resolve this conversation if no objection is raised for 24 hours. Thanks.